shared/util/saga.tsx

```diff
@@ -104,6 +104,37 @@ interface ChainAction {
     ) => ChainActionReturn,
     loggerTag?: string
   ): IterableIterator<any>
+  <AT extends [ActionTypes, ActionTypes, ActionTypes, ActionTypes, ActionTypes, ActionTypes]>(
```

Comment:
Sorry if I missed this, but I think you can reduce this type, no?

Git diff:
```
diff --git a/shared/globals.d.ts b/shared/globals.d.ts
index 6f73ee4087..ec06a5d755 100644
--- a/shared/globals.d.ts
+++ b/shared/globals.d.ts
@@ -12,6 +12,9 @@ type Unpacked<T> = T extends (infer U)[]
   ? U
   : T
 
+// Get the values of an Array. i.e. ValuesOf<["FOO", "BAR"]> => "FOO" | "BAR"
+type ValuesOf<T extends any[]> = T[number]
+
 type RequestIdleCallbackHandle = any
 type RequestIdleCallbackOptions = {
   timeout: number
diff --git a/shared/util/saga.tsx b/shared/util/saga.tsx
index 04e6a009c3..afad2f4ea5 100644
--- a/shared/util/saga.tsx
+++ b/shared/util/saga.tsx
@@ -54,83 +54,11 @@ interface ChainAction {
     handler: (state: TypedState, action: TypedActionsMap[AT], logger: SagaLogger) => ChainActionReturn,
     loggerTag?: string
   ): IterableIterator<any>
-  <AT extends [ActionTypes]>(
-    actions: AT,
-    handler: (state: TypedState, action: TypedActionsMap[AT[0]], logger: SagaLogger) => ChainActionReturn,
-    loggerTag?: string
-  ): IterableIterator<any>
-  <AT extends [ActionTypes, ActionTypes]>(
-    actions: AT,
-    handler: (
-      state: TypedState,
-      action: TypedActionsMap[AT[0]] | TypedActionsMap[AT[1]],
-      logger: SagaLogger
-    ) => ChainActionReturn,
-    loggerTag?: string
-  ): IterableIterator<any>
-  <AT extends [ActionTypes, ActionTypes, ActionTypes]>(
-    actions: AT,
-    handler: (
-      state: TypedState,
-      action: TypedActionsMap[AT[0]] | TypedActionsMap[AT[1]] | TypedActionsMap[AT[2]],
-      logger: SagaLogger
-    ) => ChainActionReturn,
-    loggerTag?: string
-  ): IterableIterator<any>
-  <AT extends [ActionTypes, ActionTypes, ActionTypes, ActionTypes]>(
-    actions: AT,
-    handler: (
-      state: TypedState,
-      action:
-        | TypedActionsMap[AT[0]]
-        | TypedActionsMap[AT[1]]
-        | TypedActionsMap[AT[2]]
-        | TypedActionsMap[AT[3]],
-      logger: SagaLogger
-    ) => ChainActionReturn,
-    loggerTag?: string
-  ): IterableIterator<any>
-  <AT extends [ActionTypes, ActionTypes, ActionTypes, ActionTypes, ActionTypes]>(
-    actions: AT,
-    handler: (
-      state: TypedState,
-      action:
-        | TypedActionsMap[AT[0]]
-        | TypedActionsMap[AT[1]]
-        | TypedActionsMap[AT[2]]
-        | TypedActionsMap[AT[3]]
-        | TypedActionsMap[AT[4]],
-      logger: SagaLogger
-    ) => ChainActionReturn,
-    loggerTag?: string
-  ): IterableIterator<any>
-  <AT extends [ActionTypes, ActionTypes, ActionTypes, ActionTypes, ActionTypes, ActionTypes]>(
-    actions: AT,
-    handler: (
-      state: TypedState,
-      action:
-        | TypedActionsMap[AT[0]]
-        | TypedActionsMap[AT[1]]
-        | TypedActionsMap[AT[2]]
-        | TypedActionsMap[AT[3]]
-        | TypedActionsMap[AT[4]]
-        | TypedActionsMap[AT[5]],
-      logger: SagaLogger
-    ) => ChainActionReturn,
-    loggerTag?: string
-  ): IterableIterator<any>
-  <AT extends [ActionTypes, ActionTypes, ActionTypes, ActionTypes, ActionTypes, ActionTypes, ActionTypes]>(
+  <AT extends ActionTypes[]>(
     actions: AT,
     handler: (
       state: TypedState,
-      action:
-        | TypedActionsMap[AT[0]]
-        | TypedActionsMap[AT[1]]
-        | TypedActionsMap[AT[2]]
-        | TypedActionsMap[AT[3]]
-        | TypedActionsMap[AT[4]]
-        | TypedActionsMap[AT[5]]
-        | TypedActionsMap[AT[6]],
+      action: TypedActionsMap[ValuesOf<AT>],
       logger: SagaLogger
     ) => ChainActionReturn,
     loggerTag?: string
```

Reply:
👍 👍  👍 I tried something like this but it wasn't working woot